whirlpool/appliance.py

```diff
@@ -86,12 +86,16 @@ async def fetch_data(self):
             return False
 
         uri = f"{self._backend_selector.base_url}/api/v1/appliance/{self._said}"
-        self._data_dict = None
+        #self._data_dict = None # Clearing the _dict causes HA to complain about missing attributes 
         async with async_timeout.timeout(30):
-            async with self._session.get(uri) as r:
+            async with self._session.get(uri,headers=self._create_headers()) as r:
```

Comment:
the session already has the correct headers, so we do not need to re-set them on the get call.

Reply:
They actually are not up to date, and are holding the OLD auth key.  When the eventsocket renews the key and tries to restart, these headers are not utilizing the new auth key (and it fails to update).

Reply:
Seems to make sense - I can't test the post parts of the code, but taking the headers out of start_http_session doesn't appear to break anything...